pkg/clusters/addons/kong/addon.go

```diff
@@ -404,7 +485,16 @@ func (a *Addon) DumpDiagnostics(ctx context.Context, cluster clusters.Cluster) (
 		if err != nil {
 			return diagnostics, err
 		}
-		diagnostics["dbless_config.json"] = b.Bytes()
+		var kongConfig struct {
+			Config string `json:"config,omitempty" yaml:"config,omitempty"`
+		}
+		err = json.Unmarshal(b.Bytes(), &kongConfig)
```

Comment:
I think you drop all of those `bytes.Buffer` shenanigans and just use: https://pkg.go.dev/encoding/json#Decoder.Decode like so

```go
json.NewDecoder(resp.Body).Decode(&kongConfig)
```